### PR TITLE
Mocha Code Coverage

### PR DESCRIPTION
**Pull request by MangeshMp**
Implemented grunt code coverage using
grunt-mocha-istanbul and istanbul

### CODE REVIEW

**Comment by MangeshMp** [comment]
Can somebody please review the code.
Thanks

**Comment by pgrodrigues** [comment]
Indentantion should be changed from spaces to tabs I believe

**Comment by MangeshMp** [comment]
Formatted package.json using webstorm IDE.

File: gruntfile.js [hunk 1]
```diff
@@ -220,7 +240,9 @@ module.exports = function (grunt) {
 	grunt.registerTask('build', ['env:dev', 'lint', 'ngAnnotate', 'uglify', 'cssmin']);
 
 	// Run the project tests
-	grunt.registerTask('test', ['env:test', 'mongoose', 'mochaTest', 'karma:unit']);
+    grunt.registerTask('test', ['test:server', 'test:client']);
```
Comment (ilanbiala) [inline gruntfile.js]: indentation

File: modules/articles/tests/server/article.server.model.tests.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+var User = require(path.resolve('./modules/users/server/models/user.server.model'));
+var Article = require(path.resolve('./modules/articles/server/models/article.server.model'));
+
 /**
  * Module dependencies.
  */
 var should = require('should'),
-	mongoose = require('mongoose'),
-	User = mongoose.model('User'),
+	mongoose = require('mongoose');
```
Comment (ilanbiala) [inline modules/articles/tests/server/article.server.model.tests.js]: revert this change

File: package.json [hunk 3]
```diff
@@ -1,97 +1,99 @@
...
-		"grunt-concurrent": "~1.0.0",
-		"grunt-mocha-test": "~0.12.1",
-		"grunt-karma": "~0.9.0",
-		"grunt-protractor-runner": "1.1.4",
-		"grunt-contrib-sass": "~0.8.1",
-		"grunt-contrib-less": "~0.12.0",
-		"load-grunt-tasks": "~1.0.0",
-		"gulp": "~3.8.9",
-		"run-sequence": "~1.0.1",
-		"gulp-rename": "~1.2.0",
```
Comment (ilanbiala) [inline package.json]: fix the spacing

File: modules/articles/tests/server/article.server.model.tests.js (hunk 2 above)
Comment (MangeshMp) [inline modules/articles/tests/server/article.server.model.tests.js]: Without this change the build fails and gives the error

  1) Article CRUD tests "before all" hook:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
      at Context.<anonymous> (modules/articles/tests/server/article.server.routes.tests.js:25:3)
Warning: Task "mochaTest:src" failed. Used --force, continuing.

Alos this fixes the issue #312 In 0.4.0 and master getting "Cannot find module when running mocha test" on grunt or gulp.

Please let me know if we should keep the change.

**Comment by MangeshMp** [comment]
Thanks @simison my webstorm was overriding editorconfig hence those indentation problem.

**Comment by ilanbiala** [comment]
@MangeshMp it's best if you squash your commits and revert all your indentation to whatever it was in each file for this PR and save that for another PR for 0.4.x.

**Comment by ilanbiala** [comment]
@MangeshMp what happened?

**Comment by MangeshMp** [comment]
I was creating a new PR and cleaning up.
I will do it today.
